mcs/db_cluster_client.go

```diff
@@ -0,0 +1,316 @@
+	return gophercloud.BuildRequestBody(opts, "")
+}
+
+// Map ...
+func (opts *DBClusterDetachConfigurationGroupOpts) Map() (map[string]interface{}, error) {
+	body, err := gophercloud.BuildRequestBody(*opts, "")
+	return body, err
+}
+
+// Map ...
+func (opts *DBClusterAttachConfigurationGroupOpts) Map() (map[string]interface{}, error) {
+	body, err := gophercloud.BuildRequestBody(*opts, "")
+	return body, err
+}
+
+// Map ...
+func (opts *DBClusterResizeVolumeOpts) Map() (map[string]interface{}, error) {
+	body, err := gophercloud.BuildRequestBody(*opts, "")
+	return body, err
+}
+
+// Map ...
+func (opts *DBClusterResizeWalVolumeOpts) Map() (map[string]interface{}, error) {
+	body, err := gophercloud.BuildRequestBody(*opts, "")
+	return body, err
+}
+
+// Map ...
+func (opts *DBClusterResizeOpts) Map() (map[string]interface{}, error) {
+	body, err := gophercloud.BuildRequestBody(*opts, "")
+	return body, err
+}
+
+// Map ...
+func (opts *DBClusterUpdateAutoExpandOpts) Map() (map[string]interface{}, error) {
+	body, err := gophercloud.BuildRequestBody(*opts, "")
+	return body, err
+}
+
+// Map ...
+func (opts *DBClusterUpdateAutoExpandWalOpts) Map() (map[string]interface{}, error) {
+	body, err := gophercloud.BuildRequestBody(*opts, "")
+	return body, err
+}
+
+// Map ...
+func (opts *DBClusterApplyCapabilityOpts) Map() (map[string]interface{}, error) {
+	body, err := gophercloud.BuildRequestBody(*opts, "")
+	return body, err
+}
+
+// Map ...
+func (opts *DBClusterGrowClusterOpts) Map() (map[string]interface{}, error) {
+	body, err := gophercloud.BuildRequestBody(*opts, "")
+	return body, err
+}
+
+// Map ...
+func (opts *DBClusterShrinkClusterOpts) Map() (map[string]interface{}, error) {
+	body, err := gophercloud.BuildRequestBody(*opts, "")
+	return body, err
+}
+
+type commonClusterResult struct {
+	gophercloud.Result
+}
+
+// CreateClusterResult ...
+type CreateClusterResult struct {
+	commonClusterResult
+}
+
+// GetClusterResult ...
+type GetClusterResult struct {
+	commonClusterResult
+}
+
+// ClusterActionResult ...
+type ClusterActionResult struct {
+	gophercloud.ErrResult
+}
+
+func (r commonClusterResult) Extract() (*DBClusterResp, error) {
+	var c *DBClusterRespOpts
+	err := r.ExtractInto(&c)
+	if err == nil {
+		return c.Cluster, nil
+	}
+	return nil, err
+}
+
+var dbClustersAPIPath = "clusters"
+
+// DBClusterCreate ...
+func DBClusterCreate(client DatabaseClient, opts OptsBuilder) (r CreateClusterResult) {
+	b, err := opts.Map()
+	if err != nil {
+		r.Err = err
+		return
+	}
+	var result *http.Response
+	reqOpts := getRequestOpts(200)
+	result, r.Err = client.Post(baseURL(client, dbClustersAPIPath), b, &r.Body, reqOpts)
+	if r.Err == nil {
+		r.Header = result.Header
+	}
+	return
+}
+
+// DBClusterGet ...
+func DBClusterGet(client DatabaseClient, id string) (r GetClusterResult) {
+	var result *http.Response
+	result, r.Err = client.Get(getURL(client, dbClustersAPIPath, id), &r.Body, nil)
```

Comment:
All GET requests should have retries. For examples, get /instance/.../databases - may be successful after several retries.

Reply:
Lets do it in a separate pr to make common solution inside mcs provider. FYI we are going to adopt OS provider for IaaS services during Q3, so this would be (and should be) a preparation task for that activity.